### PR TITLE
Adds uniqueness constraint on paper-SNP connection. 

### PR DESCRIPTION
**Pull request by philippbayer**
Replaces rake task by SQL command.

See #202

### CODE REVIEW

**Comment by tsujigiri** [comment]
I feel like this should be a migration instead of a rake task. We won't do this more than once.

File: db/structure.sql [hunk 1]
```diff
@@ -1696,59 +1864,52 @@ CREATE INDEX index_friendly_id_slugs_on_sluggable_type ON friendly_id_slugs USIN
...
-
-CREATE UNIQUE INDEX index_genotypes_by_snp_on_snp_name ON genotypes_by_snp USING btree (snp_name);
-
-
---
--- Name: index_snp_references_on_paper_id_and_paper_type; Type: INDEX; Schema: public; Owner: -; Tablespace: 
+-- Name: index_snps_chromosome_position; Type: INDEX; Schema: public; Owner: -; Tablespace: 
 --
 
-CREATE INDEX index_snp_references_on_paper_id_and_paper_type ON snp_references USING btree (paper_id, paper_type);
```
Comment (tsujigiri) [inline db/structure.sql]: Woops!

Comment (philippbayer) [inline db/structure.sql]: So the uniqueness constraint somehow kills the associated index. weird.

**Comment by tsujigiri** [comment]
You lost the indexes on snp_references. By "unique constraint" I was referring to a unique index on all three columns. Maybe even as a primary key. The structure.sql also contains lots of stuff from your local db. You can fix this by checking out the current structure.sql from master, run `rake db:reset db:migrate` and commit the resulting structure.sql.

**Comment by philippbayer** [comment]
You mean as in

     add_index :snp_references, [:snp_id, :paper_id, :paper_type], :unique => true

and remove the line in app/snp_reference.rb?

File: db/structure.sql (hunk 1 above)
Comment (tsujigiri) [inline db/structure.sql]: The reason for this is that `SELECT ... INTO ...` doesn't care for the indexes.

**Comment by tsujigiri** [comment]
By copying the table you lose the indexes. You have to re-create them. The validation on the model can stay.

**Comment by philippbayer** [comment]
Aaaah now I got you, changing

**Comment by philippbayer** [comment]
OK, got the master structure.sql, renamed and renewed snp_references indexes

File: app/models/snp_reference.rb [hunk 2]
```diff
@@ -3,4 +3,5 @@ class SnpReference < ActiveRecord::Base
   belongs_to :snp
   belongs_to :paper, polymorphic: true
   #validates_presence_of :snp, :paper
+  validates :snp_id, :uniqueness => { :scope => [:paper_id, :paper_type]}
```
Comment (tsujigiri) [inline app/models/snp_reference.rb]: The hound is supposed to prefer the colon-syntax over the hash-rocket-syntax... Stupid dog.

**Comment by tsujigiri** [comment]
Apart from the syntax-nit-pick, looks good to go.

**Pull request merged**